File: Oidc.AuthorizationServer/Models/RedirectUri.cs
using lvl.Ontology;
using lvl.Ontology.Authorization;
using lvl.Ontology.Conventions;
using System.ComponentModel.DataAnnotations;

namespace lvl.Oidc.AuthorizationServer.Models
{
    /// <summary>
    ///     Specifies allowed URIs to return tokens or authorization codes to.
    /// </summary>
    [Schema("oidc")]
    [HiddenFromApi]
    public class RedirectUri : Entity, IAggregateScope<ClientEntity>
    {
        /// <summary>
        ///     A url which is allowed.
        /// </summary>
        [Required]
        public string Name { get; set; }
    }
}

File: oidc/src/models/redirect-uri.ts
import { ClientEntity } from './client-entity';
import { Entity, IAggregateScope, Required } from '@lvl/front-end';

export class RedirectUri extends Entity implements IAggregateScope<ClientEntity> {
    @Required() public name: string;

    constructor(options?: IRedirectUriOptions) {
        options = options || {};
        super(options);
        this.name = options!.name;
    }
}

interface IRedirectUriOptions {
    name?: string;
    id?: number;
}
